src/util/bip158.rs

```diff
@@ -137,27 +137,33 @@ impl BlockFilter {
     }
 
     /// match any query pattern
-    pub fn match_any(&self, block_hash: &BlockHash, query: &mut dyn Iterator<Item=&[u8]>) -> Result<bool, Error> {
+    pub fn match_any<'a, I>(&self, block_hash: &BlockHash, query: I) -> Result<bool, Error>
+    where
+        I: Iterator<Item = &'a [u8]>,
```

Comment:
I like this -- I wish I'd kept the PR open a bit longer. (Sorry to merge out from under you but it was a set of unoffensive cleanups and I wasn't sure how long you'd be busy.)

Reply:
No problem, can always be added later.

Reply:
Why `Borrow` instead of `AsRef` @Kixunil?

> If generic code merely needs to work for all types that can provide a reference to related type T, it is often better to use [AsRef<T>](https://doc.rust-lang.org/std/convert/trait.AsRef.html) as more types can safely implement it.

ref: https://doc.rust-lang.org/std/borrow/trait.Borrow.html

I tried both and they both work, since `AsRef` is more general I'll push a PR with that, waiting on your correction though if needed. Thanks

Reply:
#1183

Reply:
It shouldn't really matter for this one. Lately I default to `Borrow` due to correctness implications, so I automatically wrote that one.

Note that there is non-correctness case when `Borrow` works and `AsRef` doesn't: when you have `T: Sized` and want to be able to take owned `T` - usually in structs but also `Iter::Item: Borrow<T>, T: Copy`.